File: schemaorggo/tvseason.go
package schemaorggo

import "encoding/json"

// TVSeason see : https://schema.org/TVSeason
type TVSeason struct {
	CreativeWork

	typeContext

	// CountryOfOrigin see : https://schema.org/countryOfOrigin
	// The country of the principal offices of the production company or individual responsible for the movie or program.
	// types : Country
	CountryOfOrigin []*Country `json:"countryOfOrigin,omitempty"`
}

func (v TVSeason) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.CreativeWork.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.CountryOfOrigin
		if len(v.CountryOfOrigin) == 1 {
			value = v.CountryOfOrigin[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["countryOfOrigin"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v TVSeason) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "TVSeason"

	return data, nil
}

func (v TVSeason) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/tv_season.php
<?php

class TVSeason extends CreativeWork implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'TVSeason';
	
	/**
	 * The country of the principal offices of the production company or individual responsible for the movie or program.
	 * see : https://schema.org/countryOfOrigin
	 * @var \Country|\Country[]
	 */
	public var $country_of_origin;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'TVSeason'
		);
		
		$serialized = so_json_serialize( $this->country_of_origin );
		if ( ! empty( $serialized ) ) {
			$out['countryOfOrigin'] = $serialized;
		}
		
		return $out;
	}
}
